src/group_theory/free_group.lean

```diff
@@ -863,4 +884,64 @@ fintype.subtype (list.to_finset $ red.enum L₁) $
 
 end reduce
 
+section metric
+
+variable [decidable_eq α]
+
+/-- The length of reduced words provides a norm on a free group. --/
+def norm (x : free_group α) : nat := x.to_word.length
+
+lemma norm_inv_le (x : free_group α) : norm x⁻¹ ≤ norm x :=
+begin
+  let w := x.to_word,
+ 
+  calc norm x⁻¹ = norm ((mk (x.to_word))⁻¹) : by simp only [to_word.mk]
+       ... = norm (mk (inv_rev w)) : by simp only [←inv_mk]
+       ... ≤ (inv_rev w).length : by exact to_word.le_length
```

Comment:
```suggestion
       ... ≤ (inv_rev w).length : to_word.le_length
```

Reply:
I think that we should have a lemma saying that `(x⁻¹).to_word = inv_rev x.to_word` (probably, not in `mathlib` yet because you're adding `inv_rev` in this PR). Then `norm_inv` will become trivial.

Reply:
`by exact` is deleted in 169439d1217979e956f2eb4a671385cf149a6633.

Reply:
a59a0a266df510c007f4aedb7dfb55079b6b8248 includes a lemma saying that `(x⁻¹).to_word = inv_rev x.to_word` and in the process I added some additional lemmas, like the fact that `inv_rev` is involutive.